s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/SourceRecordIterator.java

```diff
@@ -67,14 +68,16 @@ public final class SourceRecordIterator implements Iterator<List<AivenS3SourceRe
 
     private final OutputWriter outputWriter;
 
+    private final FileReader fileReader; // NOPMD
```

Comment:
Out of interest what was the PMD error on this entry?

Reply:
It could be converted as a local variable in the constructor, but then I found it hard to write the unit tests mocking this reader for this iterator class.